### PR TITLE
Log request and response bodies in RequestLoggingInterceptor via Thre…

### PR DESCRIPTION
**Pull request by isafan**
…adLocal

### CODE REVIEW

File: misk/src/test/kotlin/misk/web/interceptors/RequestLoggingInterceptorTest.kt [hunk 1]
```diff
@@ -61,25 +73,20 @@ internal class RequestBodyLoggingInterceptorTest {
       "ExcludeBodyRequestLoggingAction principal=caller",
       "ExcludeBodyRequestLoggingAction principal=caller time=100.0 ms code=200"
     )
-    assertThat(logCollector.takeMessages(RequestBodyLoggingInterceptor::class)).isEmpty()
+    assertThat(logCollector.takeMessages(RequestLoggingInterceptor::class)).isEmpty()
   }
 
   @Test
-  fun exceptionThrown() {
```
Comment (mightyguava) [inline misk/src/test/kotlin/misk/web/interceptors/RequestLoggingInterceptorTest.kt:68]: why did you shift the order of these tests?

Comment (isafan) [reply misk/src/test/kotlin/misk/web/interceptors/RequestLoggingInterceptorTest.kt:68]: Thought the ordering made more sense, I can undo

File: misk/src/test/kotlin/misk/web/interceptors/RequestLoggingInterceptorTest.kt [hunk 2]
```diff
@@ -61,25 +73,20 @@ internal class RequestBodyLoggingInterceptorTest {
       "ExcludeBodyRequestLoggingAction principal=caller",
       "ExcludeBodyRequestLoggingAction principal=caller time=100.0 ms code=200"
     )
-    assertThat(logCollector.takeMessages(RequestBodyLoggingInterceptor::class)).isEmpty()
+    assertThat(logCollector.takeMessages(RequestLoggingInterceptor::class)).isEmpty()
```
[the remark this answers is not in the document]
Comment (isafan) [reply misk/src/test/kotlin/misk/web/interceptors/RequestLoggingInterceptorTest.kt:76]: Oops yes that was unnecessary

File: misk/src/main/kotlin/misk/web/interceptors/RequestBodyLoggingInterceptor.kt [hunk 3]
```diff
@@ -5,61 +5,46 @@ import misk.ApplicationInterceptor
...
-
     val principal = caller.get()?.principal ?: "unknown"
 
-    logger.info { "${action.name} principal=$principal request=${chain.args}" }
+    bodyCapture.set(RequestResponseBody(chain.args, null))
 
     try {
       val result = chain.proceed(chain.args)
-      logger.info { "${action.name} principal=$principal response=$result" }
+      bodyCapture.set(RequestResponseBody(chain.args, result))
```
[the remark this answers is not in the document]
Comment (isafan) [reply misk/src/main/kotlin/misk/web/interceptors/RequestBodyLoggingInterceptor.kt:47]: I added a null check for request and response. It only appends the body to the log message if [1] includesBody is set to true and [2] if the request/response is non-null

File: misk/src/main/kotlin/misk/web/interceptors/RequestLoggingInterceptor.kt [hunk 4]
```diff
@@ -57,21 +62,37 @@ class RequestLoggingInterceptor internal constructor(
...
     } catch (t: Throwable) {
+      maybeLogRequestResponseBody(principal)
       logger.info { "${action.name} principal=$principal time=$stopwatch failed" }
       throw t
     }
   }
+
+  fun maybeLogRequestResponseBody(principal: String) {
+    if (includeBody) {
+      val body = bodyCapture.get()
```
[the remark this answers is not in the document]
Comment (isafan) [reply misk/src/main/kotlin/misk/web/interceptors/RequestLoggingInterceptor.kt]: Added a null check if it's not there

File: misk/src/main/kotlin/misk/web/interceptors/RequestBodyLoggingInterceptor.kt [hunk 5]
```diff
@@ -5,61 +5,46 @@ import misk.ApplicationInterceptor
...
+    private val bodyCapture: RequestResponseCapture
   ) : ApplicationInterceptor.Factory {
     override fun create(action: Action): ApplicationInterceptor? {
-      val logRequestResponse = action.function.findAnnotation<LogRequestResponse>() ?: return null
-      require(0.0 < logRequestResponse.sampling && logRequestResponse.sampling <= 1.0) {
-        "${action.name} @LogRequestResponse sampling must be in the range (0.0, 1.0]"
-      }
-      if (!logRequestResponse.includeBody) {
-        return null
-      }
```
Comment (mightyguava) [inline misk/src/main/kotlin/misk/web/interceptors/RequestBodyLoggingInterceptor.kt:39]: I think the block above should be kept. If body sampling isn't enabled, then you shouldn't install this interceptor

**Pull request merged**